### DISCUSSION TITLE
Why is the Y-axis inverted?

### DISCUSSION THREAD

**Discussion by Criper98** [UglyToad/PdfPig#650] [Q&A] [answered]
Why is the Y-axis inverted when creating a document?
I'm referring to the fact that the numerical value of the Y-axis starts from 0 at the bottom of the page and increases upwards.
On the other hand, the X-axis follows the standard convention, from left to right.

            PdfDocumentBuilder builder = new PdfDocumentBuilder();

            PdfPageBuilder page = builder.AddPage(PageSize.A4);

            PdfDocumentBuilder.AddedFont font = builder.AddStandard14Font(Standard14Font.Helvetica);

            for (int i = 0; i < page.PageSize.Height; i++)
            {
                if (i % 100 == 0)
                    page.AddText(i.ToString(), 10, new PdfPoint(50, i), font);
            }

            for (int i = 0; i < page.PageSize.Width; i++)
            {
                if (i % 100 == 0)
                    page.AddText(i.ToString(), 10, new PdfPoint(i, 50), font);
            }

            File.WriteAllBytes(FilePath, builder.Build());

[TEST.pdf](https://github.com/UglyToad/PdfPig/files/11786340/TEST.pdf)

Is there a way to make the Y-axis start from 0 at the top of the page and increase in value downwards?

**Comment by EliotJones** [comment, marked as answer]
Hi @Criper98 this reflects the standard coordinate system from the PDF spec, it's a bit annoying since it is different from almost everything else but is typical for how PDF coordinates work:

> The positive x axis extends horizontally to the right and the positive y axis vertically upward, as in standard mathematical prac-
tice (subject to alteration by the Rotate entry in the page dictionary). The length of a unit along both the x and y axes is set by the UserUnit entry (PDF 1.6) in the page dictionary (see Table 3.27). If that entry is not present or supported, the default value of 1 ⁄ 72 inch is used. This coordinate system is called default user space.
